addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4206,6 +4206,67 @@ QUnit.test('auto-focus composer on opening thread', async function (assert) {
     );
 });
 
+QUnit.test('mark channel as seen if last message is visible when switching channels [REQUIRE FOCUS]', async function (assert) {
+    assert.expect(1);
+
+    // channel expected to be found in the sidebar, with the expected message_unread_counter
+    // and a random unique id that will be referenced in the test
+    this.data['mail.channel'].records.push(
+        { id: 10, message_unread_counter: 1, name: 'Bla' },
+        { id: 11, message_unread_counter: 1, name: 'Blu' },
+    );
+    this.data['mail.message'].records.push({
+        body: 'oldest message',
+        channel_ids: [10],
+        id: 10,
+    }, {
+        body: 'newest message',
+        channel_ids: [11],
+        id: 11,
+    });
+    await this.start();
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="${
+                this.env.models['mail.thread'].findFromIdentifyingData({
+                    id: 11,
+                    model: 'mail.channel',
+                }).localId
+            }"]
+        `).click()
+    );
+    await afterNextRender(() => {
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="${
+                this.env.models['mail.thread'].findFromIdentifyingData({
+                    id: 10,
+                    model: 'mail.channel',
+                }).localId
+            }"]
+        `).focus();
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="${
+                this.env.models['mail.thread'].findFromIdentifyingData({
+                    id: 10,
+                    model: 'mail.channel',
+                }).localId
+            }"]
+        `).click();
+    });
+    assert.doesNotHaveClass(
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="${
+                this.env.models['mail.thread'].findFromIdentifyingData({
+                    id: 10,
+                    model: 'mail.channel',
+                }).localId
+            }"]
+        `),
+        'o-unread',
+        "sidebar item of channel ID 10 should not longer be unread"
+    );
```

Comment:
Do we really need to do 10 -> 11 -> 10 (check 10) in this test to make sense?
Can't we just 10 -> 11 (check 11)?

Reply:
It's not clear to me while we do all of those clicking around, and there's no comment or assertion to help.

Reply:
We do 11 -> 10, no channel is selected initially so we need to enable 11 and then 10

Reply:
Can't we use activeId to select 11 by default to avoid this step and then directly switch to 10?

Reply:
I don't know, do you have an example where it's used ?

Reply:
In business code it's just setting `initActiveId` but it appears we never test it. I guess this would require changing `start` helper to allow passing this value at create of discuss, so that, when `openInitThread` is called at init, it would read this value instead of "no thread".

Reply:
I found it actually:

```    await this.start({
        discuss: {
            params: {
                default_active_id: 'mail.box_moderation',
            },
        },
    });

```

Reply:
By the way, don't use `default_active_id`. Should only be used as data of the action.
Prefer `discuss.context.active_id` instead.

_(yes, we made this small mistake in most of our tests)_